src/include/OpenImageIO/ustring.h

```diff
@@ -1154,6 +1150,13 @@ template<> struct hash<OIIO::ustringhash> {
         return static_cast<std::size_t>(u.hash());
     }
 };
+
+
+// Force C++ to understand that ustring and ustringhash are trivially copyable
+// (i.e. memcpy from one ustring to another is fine).
```

Comment:
It would be good to mention the aspects that prevent c++ from just figuring this out. It seems that to make it trivially copyable, we'd have to `= default` the copy assignment operator as well as the destructor here. After that is_trivially_copyable will do the right thing by itself.

Is there any issue with just doing the following:
```
    /// ustring destructor.
    ~ustring() noexcept = default;

    /// Assign a ustring to another ustring.
    constexpr ustring& operator=(const ustring& str) = default;
```

It seems to work locally, and on a reduced repro in c++14 mode on godbolt, but it would mean changing the copy-assignment operator for both types to not return a `const` reference. That's not a typical copy-assignment setup and `=default` doesn't like that return value.  Would that break ABI maybe?

Reply:
Typical copy-assignment returns a mutable reference? Did I just miss this all along?

I don't mind doing this instead.

Reply:
I tink it shouldn't break ABI because these are all inline functions.

Reply:
But in any case, any of these changes that would break ABI will not be backported to the current release branch.

Reply:
I haven't been able to make this work. Making these two default (and changing the return value of = and assign to be non-const, which I guess is the usual way) still doesn't make this trivially copyable, on any of our CI platforms. I'll noodle with it a little more and see if I can figure out why. @jessey-git can you please share your godbolt link so I can try to figure how what I'm doing is different?

Reply:
This is what I used to test: https://godbolt.org/z/sYn6jdh3G

Reply:
Aha, my mistake.